android/src/toga_android/app.py

```diff
@@ -257,6 +257,13 @@ def get_screens(self):
         screen_list = display_manager.getDisplays()
         return [ScreenImpl(self, screen) for screen in screen_list]
 
+    ######################################################################
+    # App state
+    ######################################################################
+
+    def get_dark_mode_state(self):
+        raise NotImplementedError()
```

Comment:
It would be more consistent to use `self.factory.not_implemented("dark mode state")`, and return None directly (and similar for the other not-implemented backends)